lib/actions/addClass.js

```diff
@@ -0,0 +1,30 @@
+class Add {
+  constructor(){
+    this.success = false;
+  }
+
+  validator(){}
+
+  async execute(arg, project, db = {store(){}}){ 
+    this.success = false;
+    project.links[arg[0]] = arg[1];
+    try {
+      this.success = true;
+      await db.store(project)
+    } catch(err) {
+        this.success = false;
```

Comment:
Interesting idea. I like that you're looking for ways to expose the results of your code. Once this method returns a `Promise` the `then` component of that promise will give us success, while `catch` is failure. 

In the future, if you want a "status code" like `success`, consider returning that value instead of setting state on the class. Setting state on the class means you need to maintain it. Can you run `execute` over and over? No need to maintain all that.

Reply:
✋ I had a hand in this, @cromwellryan. We were exploring _command query separation_ earlier on when this was still looking like a OO class. Now that it's taking on more of a functional feel I agree that this is too much. https://martinfowler.com/bliki/CommandQuerySeparation.html